src/app/beer_garden/garden.py

```diff
@@ -167,6 +169,11 @@ def create_garden(garden: Garden) -> Garden:
         The created Garden
 
     """
+    spec = YapconfSpec(config._SPECIFICATION)
+    garden.connection_params = spec.load_config({"entry": garden.connection_params})[
```

Comment:
It's ultimately no different, but you can actually load the `_ENTRY_SPEC` directly to avoid the slightly awkward way this reads with the `["entry"]` bit.

```python
spec = YapconfSpec({"entry": config._ENTRY_SPEC})
garden.connection_params = spec.load_config(garden.connection_params)
```

I don't love that we're referencing the specification constants here, as the leading `_` in the name implies that those are intended to be for internal use in config.py only.  I think it's worth having a conversation about this, but given what you're trying to accomplish I think what you are doing is probably the most correct option available to you.

Reply:
I actually tried using `_ENTRY_SPEC`, but Yapconf has a strange requirement that dict that you pass in has a dict for all its key-value pairs. It definitely makes sense to just use `_ENTRY_SPEC`, but Yapconf doesn't allow it, so some bypassing has to be done with this.

Also, I understand the concern with `_`. The only place I could find a reliable source for Garden config defaults was here, though. There may be something that presents this spec info publicly, though - I will BOL for that.